src/lib/common/JsonHelper.cpp

```diff
@@ -0,0 +1,99 @@
+/*
+*
+* Copyright 2013 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms
+* of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Orion dev team
+*/
+#include "common/JsonHelper.h"
+
+std::string toJsonString(const std::string& input)
+{
+  std::ostringstream ss;
+  ss << '"';
+  for (std::string::const_iterator iter = input.begin(); iter != input.end(); ++iter) {
+        switch (*iter) {
+            case '\\': ss << "\\\\"; break;
+            case '"': ss << "\\\""; break;
+            case '/': ss << "\\/"; break;
+            case '\b': ss << "\\b"; break;
+            case '\f': ss << "\\f"; break;
+            case '\n': ss << "\\n"; break;
+            case '\r': ss << "\\r"; break;
+            case '\t': ss << "\\t"; break;
+            default: ss << *iter; break;
+      }
+  }
+  ss << '"';
+  return ss.str();
+}
```

Comment:
I think this is not a library function, and comparing with strcpy is not the right approach. It's a helper function for a little class for building a string of JSON. We cannot do every function we code as a library function. Sometimes we need to fit the specific need, not save the world

As all our cases for this use dynamic string coming from the user, basically you are trading space for speed, but "at least x10 faster"? why not x2, x1.1 or x1.01? Without measures it is sci-fi. And also the overall effect over the performance of Orion without measurement it is unknown (Sometimes the results in perform field can be counter-intuitive, these micro-benchmarks are widely discouraged)

What it is sure is your approach makes the code more unreadable, error-prone, and uncomfortable to write

The basic algorithm is the same, and the original implementation is very normal, nothing weird, so you has changed the design of the function and the callers doing another thing not functionally equivalent. Not improving the function, but another function no interchangeable at all. Please, don`t call "extremely inefficient" such code just because you think C++ is "extremely inefficient". I would prefer you use it if the complexity is O(n^2) instead of a O(n·logn) or O(n). Talking about uniformity, I don't think all the code (nor the most part) follow the design of pure C library function ... Happy to see my PRs are source of inspiration and creativity

By the way, your approach could be implemented in C++ with iterators and char sequences. It has nothing to do with plain C but with the strategy of allocating all the double of memory at first  or trying waste less memory with a smart and highly optimized implementation of the implemetarors of the STL. It's about speed vs space. But we should check the overall effect ...